.github/workflows/php.yml

```diff
@@ -0,0 +1,64 @@
+name: PHP Composer
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        php-version:
+          - "8.0"
+          - "8.1"
+
+        include:
+          - os: ubuntu-latest
+            php-version: "8.0"
+            compiler: jit
+
+          - os: ubuntu-latest
+            php-version: "8.1"
+            compiler: jit
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Install PHP with extensions
+      uses: shivammathur/setup-php@v2
+      with:
+        php-version: ${{ matrix.php-version }}
+        extensions: memcached
+        coverage: none
+
+    - name: Setup Memcached server
+      uses: niden/actions-memcached@v7
+
+    - name: Validate composer.json and composer.lock
+      run: composer validate --strict
+
+    - name: Cache Composer packages
+      id: composer-cache
+      uses: actions/cache@v2
+      with:
+        path: vendor
+        key: ${{ runner.os }}-php-${{ hashFiles('**/composer.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-php-
+
+    - name: Install dependencies
+      run: composer install --prefer-dist --no-progress
+
+    - name: Run tests with phpunit
+      run: ./vendor/bin/phpunit tests
+
+    - name: Static Analysis
+      run: ./vendor/bin/psalm --update-baseline --no-cache
```

Comment:
Why did we add `update-baseline` here?

Reply:
I think we want this to fail if the baseline changes. But I didn't actually check that that's what happens here. Might need to manually return a non-zero exit code if there's a diff.

Reply:
I changed this back 86c1a7b